pages/[report]/[fight]/index.tsx

```diff
@@ -1,118 +1,94 @@
-import React from 'react';
-import { NextPageContext } from 'next/types';
+import React, { useEffect } from 'react';
 import styled from 'styled-components';
-import dateFormat from 'dateformat';
+import { GetServerSideProps } from 'next';
+import axios from 'axios';
 import { useRouter } from 'next/router';
-import { GetBossFightsToLog } from '../../../mocks/getBossFightsToLog';
-
-interface IFight {
-  fight: {
-    difficulty: number;
-    name: string;
-    id: number;
-    kill: string;
-    fightPercentage: number;
-    friendlyPlayers: [number];
-  };
-  report: {
-    endTime: number;
-    startTime: number;
-    guild: { name: string; id: number; faction: { name: string } };
-  }
-}
+import { Simulate } from 'react-dom/test-utils';
+import {
+  IPlayerDetails, ISingleReport,
+} from '../../../interfaces';
+import NavBar from '../../../components/navbar';
+import LogMetaData from '../../../components/navbar/logMetaData';
+import FightMetaData from '../../../components/navbar/fightMetaData';
+import FeatureFilter from '../../../components/features/featureFilter';
+import PlayerTypeList from '../../../components/player/playerTypeList';
 
 const Main = styled.div`
-  padding: 4rem 0;
   flex: 1;
-  display: flex;
-  flex-direction: column;
 `;
 
-const CharacterContainer = styled.div`
+const Content = styled.div`
+  margin-top: 5rem;
   display: flex;
-  flex-direction: column;
-  margin: 0 4rem 0 4rem;
-`;
+  flex-direction: row;
+  padding: 0 5rem 0  5rem;
 
-const CharacterInfo = styled.div`
-  margin: 0.5rem 0 0.5rem 0;
-  cursor: pointer;
+  @media(max-width: 1200px){
+    flex-direction: column;
+  }
 `;
 
-const EncounterContainer = styled.div`
+const PlayerDetail = styled.div`
   display: flex;
-  flex-direction: row;
-  justify-content: space-evenly;
-  margin-bottom: 2rem;
+  flex-display: row;
+  height: 100vh;
 `;
 
-interface IUpdateQueryParam {
-  id: string,
-  value: number;
+interface IDisplayFight {
+  singleReport: ISingleReport
+  playerDetails: IPlayerDetails
 }
 
-const Fight = ({ fight, report }: IFight) => {
-  console.log('loggi ', fight);
+const Fight = ({ singleReport, playerDetails }: IDisplayFight) => {
   const router = useRouter();
+  const selectPlayer = (id: number) => {
+    router.query.playerID = id.toString();
+    router.push(router);
+  };
+
+  useEffect(() => {
+    const playerContainer = document.getElementById('playerDetails');
```

Comment:
you only need this if you have an id, therefore Id think using guard clauses would help out here:

```
const playerContainer = document.getElementById('playerDetails');
console.log(playerContainer);
    if (router.query.playerID !== undefined && playerContainer !== null) {
      playerContainer.scrollIntoView({
        behavior: 'smooth',
      });
    }
```

=> 

```
// maybe checking for undefined is better here, as the playerId could potentially be 0, right? but that case is edgier than emo-kids these days
if (!router.query.playerID){
  return;
};
const playerContainer = document.getElementById('playerDetails');
if (!playerContainer) {
  return;
}
playerContainer.scrollIntoView({
  behavior: 'smooth',
});
```

Reply:
You're right.
That particuallar case there was a 'hacky'ish solution right in the end of my hackingspree. I'll refactor it in the next iteration